etcdmain/gateway.go

```diff
@@ -81,6 +85,37 @@ func startGateway(cmd *cobra.Command, args []string) {
 			os.Exit(1)
 		}
 		plog.Infof("discovered the cluster %s from %s", eps, gatewayDNSCluster)
+
+		// confirm TLS connections are good
+		tlsInfo := transport.TLSInfo{
+			TrustedCAFile: gatewayCA,
+			ServerName:    gatewayDNSCluster,
+		}
+		t, err := transport.NewTransport(tlsInfo, 5*time.Second)
+		if err != nil {
+			plog.Fatalf("could not create transport (%v)", err)
+		}
+		for _, ep := range eps {
+			if !strings.HasPrefix(ep, "https://") {
+				if gatewayCA != "" {
+					plog.Errorf("rejecting insecure SRV record %q", ep)
```

Comment:
> A SRV record could be forged to have an http endpoint to bypass tls

Yea... We also need this for etcdctl? For peer, we can infer that from peer cert/key.